server/market/routers_test.go

```diff
@@ -589,14 +611,29 @@ func TestLimit(t *testing.T) {
 
 	ensureErr := makeEnsureErr(t)
 
+	oRig.auth.sent = make(chan *msgjson.Error, 1)
+	defer func() { oRig.auth.sent = nil }()
+
 	sendLimit := func() *msgjson.Error {
 		msg, _ := msgjson.NewRequest(reqID, msgjson.LimitRoute, limit)
-		return oRig.router.handleLimit(user.acct, msg)
+		err := oRig.router.handleLimit(user.acct, msg)
+		if err != nil {
+			return err
+		}
+		// wait for the async success (nil) / err (non-nil)
+		return <-oRig.auth.sent
```

Comment:
Since the latency queue result is async, the return from `handleLimit` is only part of what we are checking, hence pulling the `*msgjson.Error` from the new `sent` channel.